folly/experimental/exception_tracer/StackTrace.h

```diff
@@ -85,18 +98,12 @@ class StackTraceStack {
   // In debug mode, we assert that we're in zero-initialized memory by
   // checking that the two guards around top_ are zero.
   void checkGuard() const {
-#ifndef NDEBUG
-    assert(guard1_ == 0 && guard2_ == 0);
-#endif
+    if (kIsDebug) {
+      assert(guard1_ == 0 && guard2_ == 0);
+    }
   }
 
-#ifndef NDEBUG
-  uintptr_t guard1_;
-#endif
   Node* top_;
-#ifndef NDEBUG
-  uintptr_t guard2_;
```

Comment:
This is tricky because the intention is for the two guards to surround `top_`. If they are both in the base class, then they both precede `top_` rather than surrounding `top_`.

This may be solved by having three base classes....

Reply:
I was worried about the ordering of the NSDM when I made the base class. Thanks for the confirmation of my suspicion. Not ideal to have the three base classes, but I do prefer this over the preprocessor still. As such, made the change; I don't see many alternatives to force a well-specified ordering of the NSDM in a conditional manner.